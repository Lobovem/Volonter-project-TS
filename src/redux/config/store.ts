import { configureStore } from '@reduxjs/toolkit';
import contentSlice from '../slice/contentSlice';

export const store = configureStore({
  reducer: {
    content: contentSlice,
  },
});

// Выведение типов `RootState` и `AppDispatch` из хранилища
export type RootState = ReturnType<typeof store.getState>;
// Выведенные типы: {posts: PostsState, comments: CommentsState, users: UsersState}
export type AppDispatch = typeof store.dispatch;
