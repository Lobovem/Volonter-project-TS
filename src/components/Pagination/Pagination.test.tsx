import { render, screen } from '@testing-library/react';
import { Pagination } from './Pagination';
import { Provider, useSelector } from 'react-redux';
import { store } from '../../redux/config/store';
import { BrowserRouter as Router } from 'react-router-dom';


jest.mock('react-redux', () => {
  const originalModule = jest.requireActual('react-redux');
  return {
    ...originalModule,
    useDispatch: jest.fn(),
    useSelector: jest.fn(),
  };
});

describe('Pagination', () => {
  beforeEach(() => {
    jest.clearAllMocks();
  });

  const data = [{ id: 1, title: 'Новина', desc: 'News' }];

  test('renders Pagination component', () => {
    (useSelector as jest.Mock).mockReturnValue(data);
    const { container } = render(
    <Provider store={store}>
    <Router>
    <Pagination itemsPerPage={5}/>
    </Router>
  </Provider>);
  
    expect(container).toMatchSnapshot();
  });

  test('search tag h3 (decor)', () => {
        (useSelector as jest.Mock).mockReturnValue(data);
        render(
          <Provider store={store}>
          <Router>
          <Pagination itemsPerPage={5}/>
          </Router>
        </Provider>);

    const h3Element = screen.getByRole('heading', { level: 3 });
    expect(h3Element).toBeInTheDocument();
  });

  test('search link btn (2 item)', () => {
        (useSelector as jest.Mock).mockReturnValue(data);
    render(
    <Provider store={store}>
    <Router>
    <Pagination itemsPerPage={5}/>
    </Router>
  </Provider>);

    const linkBtn = screen.getAllByRole('button');
    expect(linkBtn).toHaveLength(3);
  });
});
