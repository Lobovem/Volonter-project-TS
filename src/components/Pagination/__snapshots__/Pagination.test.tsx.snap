// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pagination renders Pagination component 1`] = `
<div>
  <div
    class="news"
  >
    <h3
      class="news__decor"
    >
      Новини
    </h3>
    <div
      class="news__wrap"
    >
      <div
        class="news__inner"
      >
        <div
          class="news__imgWrap"
        >
          <a
            href="/news/1"
          >
            <img
              alt="news.jpg"
              class="news__img"
            />
          </a>
        </div>
        <div
          class="news__textBox"
        >
          <div
            class="news__textWrap"
          >
            <a
              class="news__linkAlt"
              href="/news/1"
            >
              <h2
                class="titleMain"
              >
                Новина
              </h2>
            </a>
            <p
              class="news__desc"
            >
              News
            </p>
          </div>
          <a
            class="news__link"
            href="/news/1"
          >
            Читати повністю
          </a>
        </div>
      </div>
    </div>
  </div>
  <ul
    aria-label="Pagination"
    class="pagination"
    role="navigation"
  >
    <li
      class="pagination__btn disabled"
    >
      <a
        aria-disabled="true"
        aria-label="Previous page"
        class=" disabled"
        rel="prev"
        role="button"
        tabindex="-1"
      >
        &lt;
      </a>
    </li>
    <li
      class="active"
    >
      <a
        aria-current="page"
        aria-label="Page 1 is your current page"
        class="pagination__link"
        rel="canonical"
        role="button"
        tabindex="-1"
      >
        1
      </a>
    </li>
    <li
      class="pagination__btn disabled"
    >
      <a
        aria-disabled="true"
        aria-label="Next page"
        class=" disabled"
        rel="next"
        role="button"
        tabindex="-1"
      >
        &gt;
      </a>
    </li>
  </ul>
</div>
`;
